sklearn/utils/_test_common/instance_generator.py

```diff
@@ -725,3 +736,375 @@ def _yield_instances_for_check(check, estimator_orig):
+        # TODO: replace by a statistical test, see meta-issue #16298
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+    },
+    IsolationForest: {
+        # TODO: replace by a statistical test, see meta-issue #16298
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+    },
+    KBinsDiscretizer: {
+        # TODO: fix sample_weight handling of this estimator, see meta-issue #16298
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+    },
+    KernelDensity: {
+        "check_sample_weight_equivalence": "sample_weight must have positive values"
+    },
+    KMeans: {
+        # TODO: replace by a statistical test, see meta-issue #16298
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+    },
+    KNeighborsTransformer: {
+        "check_methods_sample_order_invariance": "check is not applicable."
+    },
+    LinearRegression: {
+        # TODO: investigate failure see meta-issue #16298
+        #
+        # Note: this model should converge to the minimum norm solution of the
+        # least squares problem and as result be numerically stable enough when
+        # running the equivalence check even if n_features > n_samples. Maybe
+        # this is is not the case and a different choice of solver could fix
+        # this problem.
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+    },
+    LinearSVC: {
+        # TODO: replace by a statistical test when _dual=True, see meta-issue #16298
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+        "check_non_transformer_estimators_n_iter": (
+            "n_iter_ cannot be easily accessed."
+        ),
+    },
+    LinearSVR: {
+        # TODO: replace by a statistical test, see meta-issue #16298
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+    },
+    LogisticRegression: {
+        # TODO: fix sample_weight handling of this estimator, see meta-issue #16298
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+    },
+    MiniBatchKMeans: {
+        # TODO: replace by a statistical test, see meta-issue #16298
+        "check_sample_weight_equivalence": (
+            "sample_weight is not equivalent to removing/repeating samples."
+        ),
+    },
+    NuSVC: {
+        # This doesn't seem to fail anymore?
+        # "check_methods_subset_invariance": ("fails for the decision_function method"),
```

Comment:
by writing a test, I noticed this test doesn't actually fail. We could, in principle` now easily do this same test for all our estimators, but it would mean running the common tests twice, or to switch from `parametrize_with_checks` to `check_estimator` which I don't think is a good idea.